tests/tests-ui/cypress/support/login/commands_login.js

```diff
@@ -0,0 +1,30 @@
+Cypress.Commands.add("loginUI", (user, password) => { 
+    if(user == null){
+		user = Cypress.env('user');
+		password = Cypress.env('password');
+	}
+	login(user,password);
+	cy.get('.header-menu-buttons').should('contain', user)
+})
+
+Cypress.Commands.add("loginError", (user,password) => {
+	login(user,password);
+	cy.get('.callout').should('contain', 'Incorrect username or password');
+})
+
+
+function login(user,password) {
+	cy.visit("/login")
+	cy.get('input[name=\'username\']')
+			.type(user)
+	cy.get('input[name=\'password\']')
+			.type(password)
+	cy.get('.btn').click()
+}
+
+Cypress.Commands.add("logoutUI", () => {
+	cy.get(".icon-account_circle").click();
+	cy.get(".icon-sign-out").click();
+	cy.get('.btn').should('be.visible');
+
+})
```

Comment:
I looked at the Cypress documentation for their [recommended best practices ](https://docs.cypress.io/guides/references/best-practices.html), and they list from worst to best how to select an element. They discourage selecting by tags, classes or IDs:

> Targeting the element above by tag, class or id is very volatile and highly subject to change. You may swap out the element, you may refactor CSS and update ID’s, or you may add or remove classes that affect the style of the element.
Instead, adding the data-cy attribute to the element gives us a targeted selector that’s only used for testing.
The data-cy attribute will not change from CSS style or JS behavioral changes, meaning it’s not coupled to the behavior or styling of an element.

Additionally, it makes it clear to everyone that this element is used directly by test code. I think we should follow their recommendation here, as I can already see how every small fix in the a css or id name will lead to errors here.

Reply:
we dont have logout yes, Ill add data-cy attribute when we will have logout functionality

Reply:
Cool, but that was just one example. In tests/tests-ui/cypress/support/enums/enums_profilePage.js you also use this style